servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/DefaultGrpcClientCallFactory.java

```diff
@@ -124,11 +164,19 @@
                     serializationProvider, supportedCodings, requestClass);
             @Nullable
             final GrpcExecutionStrategy strategy = metadata.strategy();
-            final HttpResponse response = strategy == null ? client.request(httpRequest) :
-                    client.request(strategy, httpRequest);
-            return validateResponseAndGetPayload(response,
-                    serializationProvider.deserializerFor(
-                            readGrpcMessageEncoding(response, supportedCodings), responseClass));
+            try {
+                final HttpResponse response = strategy == null ? client.request(httpRequest) :
+                        client.request(strategy, httpRequest);
+                return validateResponseAndGetPayload(response,
+                        serializationProvider.deserializerFor(
+                                readGrpcMessageEncoding(response, supportedCodings), responseClass));
+            } catch (Exception all) {
+                if (all instanceof RuntimeException) {
+                    throw all;
+                } else {
+                    throw toGrpcException(all);
```

Comment:
It was mostly to avoid additional wrapping and unwrapping but since the exceptions thrown aren't easily documented it seems reasonable to only throw `GrpcStatusException`

Reply:
It would be nice if `BlockingClientCall::request(GrpcClientMetadata metadata, Req request) throws Exception;` was declared to throw a narrower exception, perhaps just `GrpcStatusException`. Throwing raw Exception doesn't encourage good handling of exceptions.